BookingService/BookingService.Testing/UserServiceTests.cs

```diff
@@ -0,0 +1,315 @@
+using BookingService.Application.DTOs;
+using BookingService.Application.Validation;
+using BookingService.Domain;
+using BookingService.Domain.Constants;
+using BookingService.Domain.Models;
+using FluentAssertions;
+using Microsoft.AspNetCore.Identity;
+using Moq;
+
+namespace BookingService.Testing
+{
+    public class UserServiceTests
+    {
+        private readonly Mock<ITokenService> _mockTokenService;
+        private readonly Mock<ICustomUserManager> _mockCustomUserManager;
+        private readonly Mock<ICustomRoleManager> _mockCustomRoleManager;
+        private readonly Mock<IMapper> _mockMapper;
+        private readonly UserService _userService;
+        private readonly LoginDtoValidator _loginDtoValidator;
+        private readonly RegisterDtoValidator _registerDtoValidator;
+
+        public UserServiceTests()
+        {
+            _mockTokenService = new Mock<ITokenService>();
+            _mockCustomUserManager = new Mock<ICustomUserManager>();
+            _mockCustomRoleManager = new Mock<ICustomRoleManager>();
+            _mockMapper = new Mock<IMapper>();
+
+            _userService = new UserService(_mockTokenService.Object, _mockCustomUserManager.Object, _mockCustomRoleManager.Object, _mockMapper.Object);
+
+            _loginDtoValidator = new LoginDtoValidator();
+            _registerDtoValidator = new RegisterDtoValidator();
+        }
+
+        [Fact]
+        public async Task LoginAsync_ShouldReturnJwtToken_WhenEmailAndPasswordAreValid()
+        {
+            //Arrange
+            var email = "test@example.com";
+            var password = "Password123*";
+            var loginDtoUser = new LoginDto { Email = email, Password = password };
+            var user = new User { Email = email, PasswordHash = password };
+            var roles = new List<string>{ Roles.Admin };
+
+            _mockCustomUserManager.Setup(userManager => userManager.FindByEmailAsync(loginDtoUser.Email)).ReturnsAsync(user);
+            _mockCustomUserManager.Setup(userManager => userManager.CheckPasswordAsync(user, loginDtoUser.Password)).ReturnsAsync(true);
+            _mockCustomUserManager.Setup(userManager => userManager.GetRolesAsync(user)).ReturnsAsync(roles);
+            _mockTokenService.Setup(tokenService => tokenService.GenerateToken(user, roles)).ReturnsAsync("validJwtToken");
+
+            //Act
+            var result = await _userService.LoginAsync(loginDtoUser);
+
+            //Assert
+            result.Should().Be("validJwtToken");
+            _mockCustomUserManager.Verify(userManager => userManager.FindByEmailAsync(loginDtoUser.Email), Times.Once());
+            _mockCustomUserManager.Verify(userManager => userManager.CheckPasswordAsync(user, loginDtoUser.Password), Times.Once);
+            _mockCustomUserManager.Verify(userManager => userManager.GetRolesAsync(user), Times.Once);
+            _mockTokenService.Verify(tokenService => tokenService.GenerateToken(user, roles), Times.Once);

```

Comment:
You did too much assertions. Your test says ShouldReturnJwtToken, so only 1 assertion is enough here. For other Verify u can create tests like `ShouldFindUserByEmail`, `ShouldCheckPassword`, etc

Reply:
BC if any of these extra assertions for your test fail, you will see failing `LoginAsync_ShouldReturnJwtToken_WhenEmailAndPasswordAreValid`, and u will think that the logic doesn't return JWT token, even though it is returning JWT